### DISCUSSION TITLE
Using Novation Launchpads with GO

### DISCUSSION THREAD

**Discussion by KenBarta** [GrandOrgue/grandorgue#1046] [Help]
### Discussed in https://github.com/GrandOrgue/grandorgue/discussions/296

<div type='discussions-op-text'>

<sup>Originally posted by **larspalo** September 16, 2021</sup>
Originally posted 2015-11-28 09:28:47.618000 by pjcomposer in GrandOrgue discussion forum on SourceForge<br><br>Earlier, I used an ordinary MIDI keyboard for GrandOrgue registration, sacrificing one manual. Since Hauptwerk has built-in support for Novation Launchpads, I figured that they would work for GO too, and recently bought a Launchpad Mini MkII to try it out.

The Launchpads are mainly MIDI keyboards with visual feedback. To light the pads, one sends key on messages with different velocity values. Like its larger siblings, the Mini has a grid of 64 rectangular pads, and two rows of 8 round pads each.

First I had to figure out the Mini&#039;s MIDI mapping and colour scheme, since they differ from the large Launchpad. The Mini has red and green LEDs with 3 levels of brightness (2-bit values), so there&#039;s 15 colours in total (plus the grey pad colour for off), though some of the colours look very similar.

I then added the MIDI values for the pads to the GO:s stops and combinations, dedicating the four 4x4 corners of the grid to each of the three manuals and the pedals. I used the top row (round pads that send MIDI controllers, not keys) for combinations, and the rightmost column (round pads) for couplers.

I mark the pads mapped to stops with a dim red colour, using the GO send off value. When a stop is drawn, the associated pad lights up in a colour reserved for that keyboard (e.g. manual I, bright red; pedals, bright green), using the on value. This works well even when the room&#039;s lighting is fairly bright.

![](http://www.parjohansson.se/bilder/LaunchpadMiniGO_small.jpg)

Here&#039;s my configuration for the Piteå organ with Comb.1 set to a Swedish romantic full organ registration (sort of). The stops are arranged as in the GO display to make them easier to remember, but I&#039;m not sure that this will work with the Barton cinema organ, which I will try next, due to the large number of stops.

Are there others out there using pad controllers for GO?

/Pär Johansson
Composer and music librarian

P.S. Thanks to the developers for a very nice app.</div>

**Comment by KenBarta** [comment]
Does anyone have experience with a LaunchPad copycat, "Midiplus"  SmartPAD? Following the above instructions, and not working at all. GO recognizes the device in MIDI devices, but I can't begin to map anything.

**Comment by ahall41** [comment]
I acquired a SubZero ControlPad64 last year. The main pads function by sending a key on/off message, which I mapped to divisional and general presets. Have you tried 'Listen for event'? Also, check what do you get when you enable 'Log MIDI events'? You will probably have to change 9x Note to 9x on (or off) toggle. 
The control buttons use SysEx messages which GO doesn't seem to understand.
Unfortunately it has stopped working and I need to send it back. Its also a bit cumbersome sat on my console, so I might exchange it for a smaller launchpad, and use it to control the sequencer instead.
